### PR TITLE
Disable sparse functions by default when building a static library

### PR DESCRIPTION
**Pull request by qjojo**
Disable sparse functions by default when building a static library.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -104,11 +104,7 @@ list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake ${ROCM_PATH}/lib
 list(APPEND CMAKE_PREFIX_PATH ${ROCM_PATH}/lib/cmake/hip /opt/rocm /opt/rocm/llvm /opt/rocm/hip)
 
 option(BUILD_SHARED_LIBS "Build hipSOLVER as a shared library" ON)
-if(BUILD_SHARED_LIBS)
-  set(BUILD_WITH_SPARSE_DEFAULT OFF)
-else()
-  set(BUILD_WITH_SPARSE_DEFAULT ON)
-endif()
+set(BUILD_WITH_SPARSE_DEFAULT OFF)
```
Comment (tfalders) [inline CMakeLists.txt]: You can probably remove BUILD_WITH_SPARSE_DEFAULT entirely. I think we only added it to manage the different defaults between the shared and static lib.

Also, do you not need to modify load_function, load_rocsparse, and load_cholmod the same way as in rocSOLVER?

**Review by jzuniga-amd — APPROVED** [review]

**Comment by qjojo** [comment]
Added the include guards from rocSOLVER and the HIPSOLVER_STATIC_LIB define

File: CMakeLists.txt (hunk 1 above)
Comment (tfalders) [reply CMakeLists.txt]: We ought to remove it from line 112 as well. I'll approve once that's done.

**Comment by qjojo** [comment]
The test failure is a segfault on sygvdx running on CUDA, I don't think that has anything to do with these changes.

**Review by cgmb — APPROVED** [review]

**Review by tfalders — APPROVED** [review]

**Comment by jmachado-amd** [comment]
> The test failure is a segfault on sygvdx running on CUDA, I don't think that has anything to do with these changes.

Yes, the segfaults on CUDA's sygvdx are happening for some time now.

**Pull request merged**